src/js/DOMUtils.js

```diff
@@ -0,0 +1,197 @@
+// Copyright 2015 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+goog.provide('axs.dom');
+
+/**
+ * Returns the nearest ancestor which is an Element.
+ * @param {Node} node
+ * @return {?Element}
+ */
+axs.dom.parentElement = function(node) {
+    if (!node)
+        return null;
+
+    var parentNode = axs.dom.composedParentNode(node);
+    if (!parentNode)
+        return null;
+
+    switch (parentNode.nodeType) {
+    case Node.ELEMENT_NODE:
+        return /** @type {Element} */ (parentNode);
+    default:
+        return axs.dom.parentElement(parentNode);
+    }
+};
+
+
+(function() {
+
+/**
+ * Generate a random ID using crypto.getRandomValues().
+ * Taken from http://stackoverflow.com/a/27747377
+ * @return {string}
+ */
+function generateId() {
+    var len = 40;
+    var arr = new Uint8Array(len / 2);
+    window.crypto.getRandomValues(arr);
+    return [].map.call(arr, function(n) { return n.toString(16); }).join("");
+}
+
+/**
+ * Returns the given Node's parent in the composed tree.
+ * @param {Node} node
+ * @return {?Node}
+ */
+axs.dom.composedParentNode = function(node) {
+    if (!node)
+        return null;
+    if (node.nodeType === Node.DOCUMENT_FRAGMENT_NODE)
+        return node.host;
```

Comment:
It might be worth noting that the result of `node.host` for a plain document fragment (e.g., one created with `document.createDocumentFragment()`) seems to be `undefined`.